src/network/gossip.rs

```diff
@@ -588,6 +591,10 @@ impl SnapchainGossip {
         );
 
         let contact_peer_id = PeerId::from_bytes(&contact_info_body.peer_id).unwrap();
+
+        self.peers
```

Comment:
@sanjayprabhu Seemingly `handle_contact_info` is not used for validator nodes, is there another preferred place to store peer information?

Reply:
Hmm, I think we should still insert into the map for validators. The intent behind the check was that the validators will never use autodiscovery and don't need to initiate connection requests to other peers.